uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -0,0 +1,485 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components'
+], function (_, moment, components) {
+  components.component('leaveCalendarMonth', {
+    bindings: {
+      contacts: '<',
+      month: '<',
+      period: '<',
+      showContactName: '<',
+      showOnlyWithLeaveRequests: '<',
+      supportData: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-calendar-month.html';
+    }],
+    controllerAs: 'month',
+    controller: ['$log', '$q', '$rootScope', 'Calendar', 'LeaveRequest', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, Calendar, LeaveRequest, sharedSettings) {
+    $log.debug('Component: leave-calendar-month');
+
+    var dataLoaded = false;
+    var calendars = {};
+    var leaveRequests = {};
+
+    var vm = this;
+    vm.currentPage = 0;
+    vm.pageSize = 20;
+    vm.visible = false;
+    vm.showContactName = vm.showContactName ? !!vm.showContactName : false;
+    vm.showOnlyWithLeaveRequests = vm.showOnlyWithLeaveRequests ? !!vm.showOnlyWithLeaveRequests : false;
+
+    vm.contactsList = contactsList;
```

Comment:
Please check the [angular style guide](https://compucorp.atlassian.net/wiki/spaces/PCHR/pages/121371880/AngularJS+Style+Guide)